lib/proxy_api/resource.rb

```diff
@@ -64,14 +64,15 @@ def parse(response)
     end
 
     # Perform GET operation on the supplied path
-    def get(path = nil, payload = {})
+    def get(path = nil, additional_headers = {})
       with_logger do
         telemetry_duration_histogram(:proxy_api_duration, :ms, method: 'get') do
           # This ensures that an extra "/" is not generated
           if path
-            resource[URI.escape(path)].get payload
+            path = "?#{URI.encode_www_form(CGI.parse(URI.parse(path).query))}" unless URI.parse(path).query.nil?
```

Comment:
wouldn't it return only the query?

didn't test it yet, though how about something like:
```ruby
    def escape_path(path)
      parsed_path = URI.parse(path)
      query = parsed_path.query
      path = "#{parsed_path.path}?#{CGI.escape(query)}" unless query.nil?
      path
    end

    # Perform GET operation on the supplied path
    def get(path = nil, additional_headers = {})
      with_logger do
        telemetry_duration_histogram(:proxy_api_duration, :ms, method: 'get') do
          # This ensures that an extra "/" is not generated
          if path
            resource[escape_path(path)].get additional_headers
          else
            resource.get additional_headers
          end
        end
      end
    end
```

Reply:
I think you are right, I update the PR with your suggestion, Thanks!